### ISSUE TITLE
Checkpoint monitor str in default Trainer

### ISSUE THREAD

**Issue by beyondyoni** [PyTorchLightning/pytorch-lightning#2912]
## ❓ Questions and Help
Hi,
In the default setting (with `checkpoint_callback=True`), when the method `configure_checkpoint_callback` is invoked by the c'ntr, the model hasn't yet been loaded to the trainer (as `run_pretrain_routine` is invoked only after `fit()`).

Thus, when calling `self.configure_checkpoint_callback(True)`, the line `train_step_only = not self.is_overridden('validation_step')` always returns `False`; even if `validation_step` was overridden in the user-class which extends `pl.LightningModule`.

This can obviously be solved by instantiating an exterior `ModelCheckpoint()`, but I guess this is not what was intended.

I'm new to this community, so I might be missing something here. 
Thanks in advance!

**Comment by rohitgr7** [comment]
Nice catch! Neither `validation_step` belongs to `self` (Trainer) here. Either it should default to `loss` or `val_loss` just like early_stopping_checkpoint.

**Comment by beyondyoni** [comment]
Thanks! :)
I think the best solution would be invoking `configure_checkpoint_callback` after `fit` is invoked, as part of `run_pretrain_routine`

**Comment by rohitgr7** [comment]
that might break things since in that case calling `trainer.fit` will initialize the checkpoint again if it is called more than once which I think is not right.

**Comment by beyondyoni** [comment]
@rohitgr7  so what's the recommended solution?
initiating the checkpoint exteriorly?
